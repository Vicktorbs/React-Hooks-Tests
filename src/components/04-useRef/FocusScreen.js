import React, { useRef } from 'react'
import '../02-useEffect/effects.css'

export const FocusScreen = () => {

    const inputRef = useRef()
    const handleClick = () => {
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input 
                ref={ inputRef }
                className="form-control mb-3"
                placeholder="Your name"
            />

            <button className="btn btn-outline-primary" onClick={ handleClick }>
                Focus
            </button>

        </div>
    )
}
