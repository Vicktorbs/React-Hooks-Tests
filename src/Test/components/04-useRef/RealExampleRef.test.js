import { shallow } from "enzyme"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Tests on file RealExampleRef', () => {
    
    const wrapper = shallow(<RealExampleRef />);
    
    test('should shoe the component <RealExampleRef /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })

    test('should show the component <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
    
})
