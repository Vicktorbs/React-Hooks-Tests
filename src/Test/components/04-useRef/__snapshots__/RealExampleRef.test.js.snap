// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on file RealExampleRef should shoe the component <RealExampleRef /> correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show / Hiden
  </button>
</div>
`;
